src/qtism/common/dom/SerializableDomDocument.php

```diff
@@ -24,27 +24,138 @@
+ * @method DOMNodeList getElementsByTagNameNS(string|null $namespace, string $localName)
+ * @method DOMElement|null getElementById(string $elementId)
+ * @method DOMNode adoptNode(DOMNode $elementId)
+ * @method append(...$nodes)
+ * @method prepend(...$nodes)
+ * @method normalizeDocument()
+ * @method renameNode(DOMNode $node, $namespace, $qualifiedName)
+ * @method DOMDocument|bool load(string $filename, ?int $options = null)
+ * @method int|false save($filename, $options = null)
+ * @method string|false saveXML(?DOMNode $node = null, int $options = null)
+ * @method bool validate()
+ * @method int|false xinclude(int $options = null)
+ * @method DOMDocument|bool loadHTML(string $source, int $options=0)
+ * @method DOMDocument|bool loadHTMLFile(string $filename, int $options=0)
+ * @method string|false saveHTML(DOMNode $node = null)
+ * @method int|false saveHTMLFile(string $filename)
+ * @method bool schemaValidate($filename, $options = null)
+ * @method bool schemaValidateSource($source, $flags)
+ * @method bool relaxNGValidate(string $filename)
+ * @method bool relaxNGValidateSource(string $source)
+ * @method bool registerNodeClass(string $baseClass, string $extendedClass)
  */
-class SerializableDomDocument extends DOMDocument
+class SerializableDomDocument
 {
-    private $xmlData;
+    private DOMDocument $dom;
+
+    public function __construct(string $version = '1.0', string $encoding = '')
+    {
+        $this->dom = new DOMDocument($version, $encoding);
+    }
+
+    public function __serialize(): array
+    {
+        return [
+            'version'  => $this->dom->xmlVersion,
+            'encoding' => $this->dom->encoding,
+            'xmlData'  => (string)$this->__toString()
+        ];
+    }
+
+    public function __unserialize(array $data): void
+    {
+        $this->dom = new DOMDocument($data['version'], $data['encoding']);
+        $this->dom->loadXML($data['xmlData']);
+    }
+
+    public function __toString(): string
+    {
+        return $this->dom->saveXML();
+    }
 
     /**
-     * @return array
+     * @throws Exception
      */
-    public function __sleep()
+    public function __call($name, $arguments)
+    {
+        if (!method_exists($this->dom, $name)) {
+            throw new Exception(__CLASS__ . 'has no method::' . $name);
+        }
+
+        return call_user_func_array([$this->dom, $name], $arguments);
+    }
+
+    public function __get($name)
     {
-        $this->xmlData = $this->saveXML();
-        return ['xmlData'];
+        if (!property_exists($this->dom, $name)) {
+            throw new Exception(__CLASS__ . 'has no property::' . $name);
+        }
+
+        return $this->dom->$name ?? null;
     }
 
-    public function __wakeup()
+    public function __set($name, $value)
     {
-        $this->loadXML($this->xmlData);
+        if (!property_exists($this->dom, $name)) {
+            throw new Exception(__CLASS__ . 'has no property::' . $name);
+        }
```

Comment:
PHP, sadly, allows dynamic properties.

This will be more aligned with a real-world behavior.

```suggestion
```